dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowCacheImpl.java

```diff
@@ -203,16 +200,16 @@ protected WorkflowTask addTask(WorkflowTask task) {
 	
 	
 	
-	protected WorkflowStep addStep(Contentlet contentlet, WorkflowStep step) {
+	protected List<WorkflowStep> addSteps(Contentlet contentlet, List<WorkflowStep> steps) {
 		 if(contentlet ==null || !UtilMethods.isSet(contentlet.getIdentifier())
-				 || step ==null || !UtilMethods.isSet(step.getId()) ){
+				 || steps ==null || !steps.isEmpty() ){
 			 return null;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)